src/ppom-storage.ts

```diff
@@ -127,80 +177,98 @@ export class PPOMStorage {
 
       syncedMetadata.push(fileMetadata);
     }
+  }
 
-    const filesInDB = await this.#storageBackend.dir();
-    for (const { name, chainId } of filesInDB) {
-      if (
-        !syncedMetadata.find(
-          (file) => file.name === name && file.chainId === chainId,
-        )
-      ) {
-        await this.#storageBackend.delete({ name, chainId });
+  const filesInState = Object.keys(stateFileStorage).map((key) => {
+    const [name, chainId] = key.split('_');
+    return { name, chainId };
+  });
+
+  for (const { name, chainId } of filesInState) {
+    if (
+      !syncedMetadata.find(
+        (file) => file.name === name && file.chainId === chainId,
+      )
+    ) {
+      if (fileStorage) {
+        delete fileStorage[`${name as string}_${chainId as string}`];
       }
-    }
 
-    this.#writeMetadata(syncedMetadata);
-    return syncedMetadata;
+      updateState({
+        fileStorage: {
+          ...fileStorage,
+        },
+      });
+    }
   }
 
-  /**
-   * Read the file from the local storage.
-   * 1. Check if the file exists in the local storage.
-   * 2. Check if the file exists in the metadata.
-   *
-   * @param name - Name assigned to storage.
-   * @param chainId - ChainId for which file is queried.
-   */
-  async readFile(name: string, chainId: string): Promise<ArrayBuffer> {
-    const metadata = this.#readMetadata();
-    const fileMetadata = metadata.find(
-      (file) => file.name === name && file.chainId === chainId,
-    );
-    if (!fileMetadata) {
-      throw new Error(`File metadata (${name}, ${chainId}) not found`);
-    }
+  updateState({
+    storageMetadata: [...syncedMetadata],
+  });
 
-    const data = await this.#storageBackend.read(
-      { name, chainId },
-      fileMetadata.checksum,
-    );
-    if (!data) {
-      throw new Error(`Storage File (${name}, ${chainId}) not found`);
-    }
+  return syncedMetadata;
+};
 
-    return data;
+/**
+ * Write the file to the local storage.
+ * 1. Write the file to the local storage.
+ * 2. Update the metadata.
+ *
+ * @param options - Object passed to write to storage.
+ * @param options.data - File data to be written.
+ * @param options.name - Name to be assigned to the storage.
+ * @param options.chainId - Current ChainId.
+ * @param options.version - Version of file.
+ * @param options.checksum - Checksum of file.
+ * @param options.readState - Controller read state function.
+ * @param options.updateState - Controller function to update state.
+ */
+export const writeFile = async ({
+  data,
+  name,
+  chainId,
+  version,
+  checksum,
+  readState,
+  updateState,
+}: {
```

Comment:
It might be better to pass whole object `fileVersionInfo` here rather then all fields `name, chainId, version, checksim`.